src/LuaPiGui.cpp

```diff
@@ -1483,78 +1519,86 @@ static int l_pigui_get_projected_bodies_grouped(lua_State *l)
 		filtered.back()._body = body;
 	}
 
-	std::vector<TSS_vector> groups;
+	struct GroupInfo {
+		Body *m_mainBody;
+		vector2d m_screenCoords; // screen coords of group
+		std::vector<Body *> m_bodies;
+		bool m_hasNavTarget;
+
+		GroupInfo(Body *b, const vector2d &coords, bool isNavTarget) :
```

Comment:
```suggestion
		GroupInfo(Body *b, const vector2d &coords, bool isNavTarget, size_t num) :
```
...where num is used to...